src/Common/tests/System/Net/Configuration.Certificates.cs

```diff
@@ -45,14 +45,24 @@ private static X509Certificate2Collection GetCertificateCollection(string certif
             {
                 // On Windows, .Net Core applications should not import PFX files in parallel to avoid a known system-level race condition.
                 // This bug results in corrupting the X509Certificate2 certificate state.
-                Assert.True(m.WaitOne(MutexTimeout), "Cannot acquire the global certificate mutex.");
-
-                var certCollection = new X509Certificate2Collection();
-                certCollection.Import(Path.Combine(TestDataFolder, certificateFileName), CertificatePassword, X509KeyStorageFlags.DefaultKeySet);
+                try
+                {
+                    Assert.True(m.WaitOne(MutexTimeout), "Cannot acquire the global certificate mutex.");
 
-                m.ReleaseMutex();
+                    var certCollection = new X509Certificate2Collection();
+                    certCollection.Import(Path.Combine(TestDataFolder, certificateFileName), CertificatePassword, X509KeyStorageFlags.DefaultKeySet);
```

Comment:
I've tested this and made it crash by forcing an exception. The second instance won't throw the exception.  I think that's because GC will still run the finalizer at some point before the process crashes.

In release this won't be a problem (the `finally` always gets executed).